### PR TITLE
python3Packages.yark: init at 1.2.3

### PR DESCRIPTION
**Pull request by MostAwesomeDude**
By request, in #209237, but also it's been on my plate for about a week to investigate. Since many folks want this, I'm allowing edits to the PR so that we can get this merged as soon as everybody is happy.

I have tested that the main executable runs. I tested the following workflow, based on the upstream README:

    $ yark new dr.fatima https://www.youtube.com/channel/UCUNtD-UN-fQHHteLHK9HgvQ
    $ yark refresh dr.fatima --skip-download
    $ yark view dr.fatima

###### Description of changes

[yark](https://github.com/Owez/yark) is a YouTube archival tool. It saves metadata and videos from YouTube into archive-ready folders.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Owez** [comment]
Author of yark here, looks like a good contrib and the hash checks out

image

File: pkgs/development/python-modules/yark/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-KMnQpEH2Z19Y0jBjqx2rZjmlle2M9bcuDCjDIljQEYY=";
+  };
+
+  propagatedBuildInputs = [
+    click colorama flask requests yt-dlp
+  ];
+
+  doCheck = false;
```
Comment (NickCao) [inline pkgs/development/python-modules/yark/default.nix:17]: Better add a note about why checks are disabled.

Comment (MostAwesomeDude) [reply pkgs/development/python-modules/yark/default.nix:19]: Done, thanks.

**Review by NickCao — APPROVED** [review]

File: pkgs/development/python-modules/yark/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  ];
+
+  # There aren't any unit tests. If test discovery runs, it will crash, halting the build.
+  # When upstream adds unit tests, please configure them here. Thanks! ~ C.
+  doCheck = false;
+
+  meta = with lib; {
+    description = "YouTube archiving made simple";
+    homepage = "https://github.com/Owez/yark";
+    license = licenses.mit;
```
Comment (ajs124) [inline pkgs/development/python-modules/yark/default.nix:24]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes requires `meta.maintainers must be set.`

Comment (NickCao) [reply pkgs/development/python-modules/yark/default.nix:24]: @MostAwesomeDude would you like to open a followup PR adding yourself as a maintainer?

**Comment by Owez** [comment]
A minor 1.2.4 change has been released also been released. Changes for a followup PR look to be:

1. `meta.maintainers`
2. Bump version

The new version's hash (non-base64) is `fdd50d0e832b4522cbe24433f42ed571a1f199f571c1d0c98472b94a501db9cf`

**Pull request merged**